django/contrib/postgres/search.py

```diff
@@ -94,11 +104,19 @@ def __init__(self, value, output_field=None, **extra):
         self.invert = extra.pop('invert', self.invert)
         super(SearchQuery, self).__init__(value, output_field=output_field)
 
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
+        if self.config:
+            if not hasattr(self.config, 'resolve_expression'):
+                self.config = Value(self.config)
+            self.config = self.config.resolve_expression(query, allow_joins, reuse, summarize, for_save)
```

Comment:
Careful, you're mutating `self.config` rather than a copy of it. If you hang onto an instance of `SearchVector` and reuse it then resolve_expression will be called on it twice. You'd be better off figuring out a way to add `config` via `self.set_source_expressions()` rather than overriding resolve_expression and `as_sql`. Even if that means using private expressions to build the components.